libs/checkpoint-redis/src/store.ts

```diff
@@ -0,0 +1,1057 @@
+    }
+  }
+
+  async listNamespaces(options?: {
+    prefix?: string[];
+    suffix?: string[];
+    maxDepth?: number;
+    limit?: number;
+    offset?: number;
+  }): Promise<string[][]> {
+    let query = "*";
+
+    try {
+      const results = await this.client.ft.search("store", query, {
+        LIMIT: { from: 0, size: 1000 }, // Get many to deduplicate
+        RETURN: ["prefix"],
+      });
+
+      // Extract unique namespaces and filter
+      const namespaceSet = new Set<string>();
+      for (const doc of results.documents) {
+        const prefix = (doc.value as unknown as StoreDocument).prefix;
+        const parts = prefix.split(".");
+
+        // Apply prefix filter if specified
+        if (options?.prefix) {
+          // Check if this namespace starts with the specified prefix
+          if (parts.length < options.prefix.length) continue;
+
+          let matches = true;
+          for (let i = 0; i < options.prefix.length; i++) {
+            if (parts[i] !== options.prefix[i]) {
+              matches = false;
+              break;
+            }
+          }
+          if (!matches) continue;
+        }
+
+        // Apply suffix filter if specified
+        if (options?.suffix) {
+          // Check if this namespace ends with the specified suffix
+          if (parts.length < options.suffix.length) continue;
+
+          let matches = true;
+          const startIdx = parts.length - options.suffix.length;
+          for (let i = 0; i < options.suffix.length; i++) {
+            if (parts[startIdx + i] !== options.suffix[i]) {
+              matches = false;
+              break;
+            }
+          }
+          if (!matches) continue;
+        }
+
+        // Apply max depth
+        if (options?.maxDepth) {
+          const truncated = parts.slice(0, options.maxDepth);
+          namespaceSet.add(truncated.join("."));
+        } else {
+          namespaceSet.add(prefix);
+        }
+      }
+
+      // Convert to array of arrays and sort
+      let namespaces = Array.from(namespaceSet)
+        .map((ns) => ns.split("."))
+        .sort((a, b) => a.join(".").localeCompare(b.join(".")));
+
+      // Apply pagination
+      if (options?.offset || options?.limit) {
+        const offset = options.offset || 0;
+        const limit = options.limit || 10;
+        namespaces = namespaces.slice(offset, offset + limit);
+      }
+
+      return namespaces;
+    } catch (error: any) {
+      if (error.message?.includes("no such index")) {
+        return [];
+      }
+      throw error;
+    }
+  }
+
+  async batch(ops: Operation[]): Promise<any[]> {
+    const results: any[] = new Array(ops.length).fill(null);
+
+    // Process operations in order to maintain dependencies
+    for (let idx = 0; idx < ops.length; idx++) {
+      const op = ops[idx];
+
+      // Execute operation based on discriminated union type
+      if ("value" in op) {
+        // PutOperation
+        const putOp = op as PutOperation;
```

Comment:
_note_: will look into exporting type guard helpers for operations

Reply:
Addressed